internal/app/server/open_saves.go

```diff
@@ -457,6 +466,186 @@ func (s *openSavesServer) Ping(ctx context.Context, req *pb.PingRequest) (*pb.Pi
 	}, nil
 }
 
+func (s *openSavesServer) CreateChunkedBlob(ctx context.Context, req *pb.CreateChunkedBlobRequest) (*pb.CreateChunkedBlobResponse, error) {
+	b := blobref.NewChunkedBlobRef(req.GetStoreKey(), req.GetRecordKey())
+	b, err := s.metaDB.InsertBlobRef(ctx, b)
+	if err != nil {
+		log.Errorf("CreateChunkedBlob failed for record (%v), blob key (%v): %v", b.RecordKey, b.Key, err)
+		return nil, err
+	}
+	return &pb.CreateChunkedBlobResponse{
+		SessionId: b.Key.String(),
+	}, nil
+}
+
+func (s *openSavesServer) UploadChunk(stream pb.OpenSaves_UploadChunkServer) error {
+	ctx := stream.Context()
+	req, err := stream.Recv()
+	if err != nil {
+		log.Errorf("Recv() returned error: %v", err)
+		return err
+	}
+	meta := req.GetMetadata()
+	if meta == nil {
+		log.Error("GetMetadata() returned nil. First message should be a ChunkMetadata.")
+		return status.Error(codes.InvalidArgument, "First message should be a ChunkMetadata.")
+	}
+	blobKey, err := uuid.Parse(meta.GetSessionId())
+	if err != nil {
+		log.Errorf("SessionId is not a valid UUID string: %v", err)
+		return status.Errorf(codes.InvalidArgument, "SessionId is not a valid UUID string: %v", err)
+	}
+
+	// Create a chunk reference based on the metadata.
+	chunk := chunkref.New(blobKey, int32(meta.GetNumber()))
+	if err := s.metaDB.InsertChunkRef(ctx, chunk); err != nil {
+		return err
+	}
+
+	writer, err := s.blobStore.NewWriter(ctx, chunk.ObjectPath())
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if writer != nil {
+			writer.Close()
+			// This means an abnormal exit, so make sure to mark the blob as Fail.
+			s.chunkRefFail(ctx, chunk)
+		}
+	}()
+
+	written := int32(0)
```

Comment:
Is there a limit on the size of a chunk? If so, this might not be large enough.

Reply:
Added a size check